File: lib/py/msgpact/internal/generation/GenerateRandomUMockCall.py
from ...internal.generation.GenerateContext import GenerateContext
from ...internal.generation.GenerateRandomUnion import generate_random_union
from ..types.VFn import VFn
from ..types.VType import VType


def generate_random_u_mock_call(types: dict[str, VType], ctx: GenerateContext) -> object:
    functions = [
        value for key, value in types.items()
        if isinstance(value, VFn) and not key.endswith('_')
    ]

    functions.sort(key=lambda fn: fn.name)

    selected_fn = functions[ctx.random_generator.next_int_with_ceiling(
        len(functions))]

    return generate_random_union(None, False, selected_fn.call.tags, ctx.copy(always_include_required_fields=False))

File: lib/java/src/main/java/io/github/telepact/internal/generation/GenerateRandomUMockCall.java
package io.github.telepact.internal.generation;

import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

import io.github.telepact.internal.types.VFn;
import io.github.telepact.internal.types.VType;

public class GenerateRandomUMockCall {

    public static Object generateRandomUMockCall(Map<String, VType> types, GenerateContext ctx) {
        List<VFn> functions = types.entrySet().stream()
                .filter(entry -> entry.getValue() instanceof VFn)
                .filter(entry -> !entry.getKey().endsWith("_"))
                .map(entry -> (VFn) entry.getValue())
                .sorted((fn1, fn2) -> fn1.getName().compareTo(fn2.getName()))
                .collect(Collectors.toList());

        VFn selectedFn = functions.get(ctx.randomGenerator.nextIntWithCeiling(functions.size()));

        return GenerateRandomUnion.generateRandomUnion(null, false, selectedFn.call.tags,
                ctx.copyWithNewAlwaysIncludeRequiredFields(false));
    }
}
